File: src/leetCodeProblems/BruteForce/FindNUniqueIntegersSumUpToZero1304.java
package leetCodeProblems.BruteForce;

/**
 * LeetCode - https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/submissions/
 *
 * TimeComplexity - O(n)
 * SpaceComplexity - O(n)
 */

import java.util.Arrays;

public class FindNUniqueIntegersSumUpToZero1304 {

    public int[] sumZero(int n) {

        int baseElement = 1;
        int currentIndex = 0;

        int remaining = n;

        int[] output = new int[n];

        while (currentIndex < n && remaining > 1) {

            output[currentIndex] = baseElement;
            currentIndex++;

            output[currentIndex] = -baseElement;
            currentIndex++;

            baseElement++;
            remaining -= 2;
        }

        return output;
    }

    public static void main(String[] args) {

        FindNUniqueIntegersSumUpToZero1304 obj = new FindNUniqueIntegersSumUpToZero1304();

        int[] output = obj.sumZero(5);
        System.out.println(Arrays.toString(output));

        int[] output1 = obj.sumZero(4);

        System.out.println(Arrays.toString(output1));
    }
}

File: src/leetCodeProblems/BruteForce/FindNUniqueIntegersSumUpToZero1304.go
package main

import "log"

/*
- LeetCode - https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/
- Time - O(n)
- Space - O(1)
*/

func sumZero(n int) []int {
	out := []int{}

	if n%2 != 0 {
		out = append(out, 0)
	}

	for i := 1; i <= n/2; i++ {
		out = append(out, i)
		out = append(out, -i)
	}
	return out
}

func main() {
	log.Println(sumZero(5))
	log.Println(sumZero(3))
	log.Println(sumZero(1))
	log.Println(sumZero(4))
}
